### DISCUSSION TITLE
什么原因导致的v2ray不可用

### DISCUSSION THREAD

**Discussion by linjinbao666** [v2fly/v2ray-core#2355] [Q&A]
背景描述：
  - v2ray服务端的配置用的是dynamic+ws，配置了6个用户，配置了`geopip:cn` rejected, 然后同样的配置在3台vps上使用
  - 一个用户的客户端配置了3个v2ray的负载均衡
  - 服务稳定运行了2年左右
  - 昨晚开始端口被封禁，换端口可以使用，重点是**3台服务器的VPS端口同时被封禁了**

原因猜测：
  - 近期加了一个新的用户，配置了一个客户端配置，通过微信发送给对方，怀疑是微信检测到的
  - 日志看到有一个用户客户端使用了全局模式，国内域名也会请求到我的服务端，
  - 怀疑是用了v2ray负载均衡的用户被检测到了

有没有熟悉这里面的机制的同学，哪个被封的原因最有可能。

**Comment by mydogshitgold** [comment]
> Recently added a new user, configured a client configuration, and sent it to the other party through WeChat, suspected to be detected by WeChat

Ha, look where they got you! If you wish share your private server and stay gray, no can do! V2Ray servers should be stay hidden from eyes other than yours.

I am not sure what kind of stimulus triggers the block, but there is certainly something you can do better according to popular beliefs and common sense.

- Never share your hidden server with others
- If you must, share a spare server
- Most importantly, NEVER EVER share your v2ray service on  the demostic messaging platform
- Use an anonymous network / messaging platform to share your server with others.
- Hiding servers behind CDN or replay can help. It can be the savior of your blocked servers
- Block access to demostic network on the server side
- Don't use proprietary clients
- Do not run the v2ray client on Android phones with demostic ROMs
- If you're paranoid, don't use Windows, get your self a Linux distribution.
- Or better yet, don't use proprietary software

This is not very relevant for hiding proxy servers, but you might want to take a look at it: https://anonymousplanet-ng.org/guide.html

BTW, if you don't have a working internet right now, GitHub codespaces can be used to proxy your network. Download [github CLI](https://github.com/cli/cli) and run `gh cs ssh -- -D 9000`, to get a working socks proxy on port :9000.
